server/core/src/actors/internal.rs

```diff
@@ -100,22 +100,55 @@ impl QueryServerWriteV1 {
         }
     }
 
-    pub(crate) async fn handle_delayedaction(&self, da: DelayedAction) {
+    pub(crate) async fn handle_delayedaction(&self, da_batch: &mut Vec<DelayedAction>) {
         let eventid = Uuid::new_v4();
         let span = span!(Level::INFO, "process_delayed_action", uuid = ?eventid);
 
+        let mut retry = false;
+
         async {
             let ct = duration_from_epoch_now();
             let mut idms_prox_write = self.idms.proxy_write(ct).await;
-            if let Err(res) = idms_prox_write
-                .process_delayedaction(da, ct)
-                .and_then(|_| idms_prox_write.commit())
-            {
-                info!(?res, "delayed action error");
+
+            for da in da_batch.iter() {
+                retry = idms_prox_write.process_delayedaction(da, ct).is_err();
+                if retry {
+                    // exit the loop
+                    break;
+                }
+            }
+
+            if let Err(res) = idms_prox_write.commit() {
+                retry = true;
+                warn!(?res, "delayed action commit error");
```

Comment:
isn't this an "OH NO LET'S BAIL" situation at this point?

Reply:
The problem here is that if we are writing a stripe of delayed actions and one of the set fails, we don't want to negatively impact the other elements of that set. So we fall back to one-at-a-time processing to allow everything else to proceed correctly even if a tad slower.

Reply:
'k, just thinking if the commit fails after an error, then something horrible's going on.

Reply:
Yeah, but even if we panic we can't do much about it I don't think. Well a panic would poison the write lock and prevent any new writes I guess ....

Reply:
I am wondering if we should make fn commit panic instead ....

Reply:
well I don't think a panic would be right but an `Err` at least :D